### PR TITLE
Storage API iteration 1

### PR DESCRIPTION
**Pull request by duhizjame**

### CODE REVIEW

File: docs/protocol/whitefox-protocol-api.yml [hunk 1]
```diff
@@ -576,7 +576,7 @@ paths:
           $ref: "#/components/responses/409"
         "500":
           $ref: "#/components/responses/500"
-  /whitefox-api/v1/storage:
```
Comment (tmnd1991) [inline docs/protocol/whitefox-protocol-api.yml:579]: [storage is an uncountable noun](https://www.oxfordlearnersdictionaries.com/definition/english/storage#:~:text=storage-,noun,where%20things%20can%20be%20kept) 😄

File: server/src/main/java/io/whitefox/api/server/StorageV1ApiImpl.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  public Response createStorage(io.whitefox.api.model.v1.generated.CreateStorage createStorage) {
+    return wrapExceptions(
+        () -> Response.status(Response.Status.CREATED)
+            .entity(Mappers.storage2api(storageService.createStorage(
+                Mappers.api2createStorage(createStorage, getRequestPrincipal()))))
+            .build(),
+        exceptionToResponse);
+  }
+
+  private Principal getRequestPrincipal() {
```
Comment (tmnd1991) [inline server/src/main/java/io/whitefox/api/server/StorageV1ApiImpl.java:30]: we should factor this "service" somewhere

**Review by tmnd1991 — CHANGES_REQUESTED** [review]
Good job 😄 minor changes requested

File: server/src/main/java/io/whitefox/core/MetastoreType.java [hunk 3]
```diff
@@ -8,7 +8,7 @@ public enum MetastoreType {
 
   public final String value;
 
-  private MetastoreType(String value) {
+  MetastoreType(String value) {
```
[the remark this answers is not in the document]
Comment (tmnd1991) [reply server/src/main/java/io/whitefox/core/MetastoreType.java:11]: everyday I learn something new 👍

File: server/src/main/java/io/whitefox/api/server/StorageV1ApiImpl.java (hunk 2 above)
Comment (duhizjame) [reply server/src/main/java/io/whitefox/api/server/StorageV1ApiImpl.java:30]: yeah, I thought of putting it under ApiUtils for now since it is a mock impl

**Review by tmnd1991 — APPROVED** [review]

**Pull request merged**